SeaPublicWebsite/ExternalServices/BreApi.cs

```diff
@@ -73,11 +73,12 @@ public static async Task<List<Recommendation>> GetRecommendationsForUserRequest(
                         return recommendations;
                     }
 
-                    return null;
+                    throw new ArgumentNullException();
```

Comment:
This isn't really what `ArgumentNullException`s are for. They are meant to be used when a function is given `null` when it shouldn't be. (Also this is still returning null below in the `catch` block)
Given that we're not sure how we'll be handling errors yet maybe just go with `throw new Exception($"BRE API returned an error: {response.StatusCode}")` for now (and something similar below)